src/ert/scheduler/job.py

```diff
@@ -141,6 +142,7 @@ async def _max_runtime_task(self) -> None:
         self.returncode.cancel()  # Triggers CancelledError
 
     async def _send(self, state: State) -> None:
+        self.state = state
```

Comment:
Maybe slightly naughty to have a state change as a bi-product of sending events. Unless we claim it is an implementation detail.

Reply:
Imo it's fine. We can rename the function to `_set_state` and have sending events be the side-effect but it's whatever.